File: assets/check.py
#!/usr/bin/env python3

import hashlib
import json
import sys

from util import calculate_checksum

payload = json.loads(sys.stdin.read().strip())
source = payload.get("source", {})

filename = source.get("filename", "")
content = source.get("content", "")

version = [
    {"sha256sum": calculate_checksum(filename, content)}
]

sys.stdout.write(json.dumps(version))

File: assets/check.js
#!/usr/bin/env node

const lib = require('./lib')

async function main () {
  try {
    const payload = await lib.populatePayloadAsync()
    const { source = {}, version } = payload
    const { filename, content } = source

    const digest = lib.calculateChecksum(filename, content)

    const versions = []
    if (version) {
      versions.push(version)
    }
    versions.push({ sha256sum: digest })

    console.log(JSON.stringify(versions))
  } catch (e) {
    console.error(e.message)
    return process.exit(1)
  }
}

main()
